workflows/sc_adaptive_over_time/example_sca_over_time.py

```diff
@@ -0,0 +1,312 @@
+        alpha=0.2,
+    )
+    ax.plot(xvalues, results.describe(var, "mean"), label="mean")
+    try:
+        ax.plot(xvalues, results.describe(var, "1%"), "--", label="1%", color="black")
+        ax.plot(xvalues, results.describe(var, "99%"), "--", label="99%", color="black")
+    except RuntimeError:
+        pass
+    ax.grid(True)
+    ax.set_ylabel(var)
+    ax.set_xlabel(r"$\rho$")
+    ax.set_title("iteration " + str(i))
+    ax.legend()
+    fig.savefig(filename)
+    plt.close()
+
+
+encoder = boutvecma.BOUTEncoder(template_input="../../models/conduction/data/BOUT.inp")
+
+sample_locations = [
+    {"variable": "T", "output_name": "T_centre", "x": 0, "y": 50, "z": 0},
+    {"variable": "T", "output_name": "T_edge", "x": 0, "y": 10, "z": 0},
+]
+decoder = boutvecma.SampleLocationBOUTDecoder(sample_locations=sample_locations)
+params = {
+    "conduction:chi": {"type": "float", "min": 0.0, "max": 1e3, "default": 1.0},
+    "T:scale": {"type": "float", "min": 0.0, "max": 1e3, "default": 1.0},
+    "T:gauss_width": {"type": "float", "min": 0.0, "max": 1e3, "default": 0.2},
+    "T:gauss_centre": {"type": "float", "min": 0.0, "max": 2 * np.pi, "default": np.pi},
+}
+actions = uq.actions.local_execute(
+    encoder,
+    os.path.abspath(
+        "../../build/models/conduction/conduction -q -q -q -q  -d . nout=200 |& tee run.log"
+    ),
+    decoder,
+)
+campaign = uq.Campaign(name="Time.", actions=actions, params=params)
+
+vary = {
+    "conduction:chi": chaospy.Uniform(0.2, 4.0),
+    # "conduction:chi": chaospy.LogUniform(0.2, 4.0),
+    "T:scale": chaospy.Uniform(0.5, 1.5),
+    "T:gauss_width": chaospy.Uniform(0.5, 1.5),
+    "T:gauss_centre": chaospy.Uniform(0.5 * np.pi, 1.5 * np.pi),
+}
+
+# sampler = uq.sampling.PCESampler(vary=vary, polynomial_order=3)  # , rule="c")
+# sampler = uq.sampling.PCESampler(vary=vary, polynomial_order=[1,1,1])#, rule="c")
+# sampler = uq.sampling.SCSampler(vary=vary, polynomial_order=3)#, rule="c")
+
+sampler = uq.sampling.SCSampler(
+    vary=vary,
+    polynomial_order=1,
+    quadrature_rule="C",
+    sparse=True,
+    growth=True,
+    midpoint_level1=True,
+    dimension_adaptive=True,
+)
+campaign.set_sampler(sampler)
+
+print(f"Computing {sampler.n_samples} samples")
+
+time_start = time.time()
+campaign.execute().collate()
+
+# results = campaign.analyse(qoi_cols=["T"])
+
+# Create an analysis class and run the analysis.
+analysis = uq.analysis.SCAnalysis(sampler=sampler, qoi_cols=["T_centre", "T_edge"])
+campaign.apply_analysis(analysis)
+plot_grid_2D(0, "grid0.png")
+
+i = 0
+sobols_error = 1e6
+error_vs_its = []
+samples_vs_its = []
+while sobols_error > 1e-3:
+    i += 1
+    refine_sampling_plan(1)
+    campaign.apply_analysis(analysis)
+    results = campaign.last_analysis
+    samples_vs_its.append(sampler.n_samples)
+
+    plot_grid_2D(i, "grid" + str(i) + ".png")
+    moment_plot_filename = os.path.join(
+        f"{campaign.campaign_dir}", "moments_centre" + str(i) + ".png"
```

Comment:
This is fine, but probably nicer to use f-strings:

```suggestion
        f"{campaign.campaign_dir}", f"moments_centre{i}.png"
```